sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -374,8 +380,46 @@ public void addIntegration(@NotNull Integration integration) {
    *
    * @param dsn the DSN
    */
-  public void setDsn(@Nullable String dsn) {
+  public void setDsn(final @Nullable String dsn) {
     this.dsn = dsn;
+
+    dsnHash = calculateHashedDsn(this.dsn);
+  }
+
+  private @Nullable String calculateHashedDsn(final @Nullable String dsn) {
+    if (dsn == null || dsn.isEmpty()) {
+      return null;
+    }
+
+    try {
+      // getInstance() method is called with algorithm SHA-1
```

Comment:
Similar to https://github.com/getsentry/sentry-cocoa/blob/e254b733f7706d4c04c02093f5ad0715ced92443/Sources/Sentry/SentryDsn.m#L31-L41 I believe